adapters/ads_adapter.py

```diff
@@ -1,22 +1,167 @@
-""" Beckhoff ADS adapter.
+"""Beckhoff ADS adapter.
 
-Downloads all beckhof ADS data and sends it to the GUI tool.
+Subscribes to Beckhof ADS variables and sends it to the GUI tool.
+
+Auto subscribe only works for DINT and LREAL for now.
+
+Example usage:
+    python ads_adapter.py --regex GVl.*temperature
 
 """
+from lognplot.client import LognplotTcpClient
+import pyads
+import time
+import ctypes
+import struct
+import datetime
+import re
+import argparse
+from collections import namedtuple
+from typing import Type
 
-# TODO: implement.
 
+class AdsClient:
+    Entry = namedtuple(
+        "Entry", ["name", "typename", "comment", "datatype", "datatype_size"]
+    )
 
+    # Some data types have same number, maybe use type string field found in __parse_ads_entries...
+    DATATYPE_MAP = {
+        17: pyads.PLCTYPE_BOOL,  # ADST_UINT8
+        17: pyads.PLCTYPE_BYTE,  # ADST_UINT8
```

Comment:
This key is duplicate, it might make sense to reverse the dictionary so that the numbers are the values and the `pyads` types are the keys. Then this mapping can be derived from that mapping.

Reply:
I was thinking of solving this by not using the ``datatype`` field of an entry. Since we only have to do this once (at ``add_device_notification``), we can also just use the ``typename`` field.